### PR TITLE
Score cleanup, add remaining offer endpoints and user endpoints, prepare pom for testing

### PR DESCRIPTION
**Pull request by Sedarius-1**

### CODE REVIEW

File: offer/src/main/java/org/shareio/backend/core/usecases/service/SearchOffersUseCaseService.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    @Override
+    public List<UUID> getOfferListMeetingCriteria(UUID userId, String title, String category, String condition, Double distance, Double score, LocalDate creationDate, String sortType) {
+        User user = getUserProfileDaoInterface.getUserDto(userId).map(User::fromDto).orElseThrow(NoSuchElementException::new);
+        List<Offer> offers = getAllOffersDaoInterface.getAllOffers().stream().map(Offer::fromDto).toList();
+
+        // FILTER
+        if (Objects.nonNull(title) && !title.isBlank()) {
+            offers = offers.stream()
+                    .filter(offer -> offer.getTitle().toUpperCase().contains(title.toUpperCase())).toList();
```
Comment (242370) [inline offer/src/main/java/org/shareio/backend/core/usecases/service/SearchOffersUseCaseService.java:37]: Change implementation to the one in getOffersByName

File: offer/src/main/java/org/shareio/backend/core/model/vo/OfferSortType.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package org.shareio.backend.core.model.vo;
+
+public enum OfferSortType {
+    CLOSEST, FURTHEST, NEWEST, OLDEST, RATED
```
Comment (242370) [inline offer/src/main/java/org/shareio/backend/core/model/vo/OfferSortType.java]: Change RATED to HIGHEST_RATED

**Review by 242370 — CHANGES_REQUESTED** [review]

File: offer/src/main/java/org/shareio/backend/core/usecases/service/SearchOffersUseCaseService.java (hunk 1 above)
Comment (Tomunek) [reply offer/src/main/java/org/shareio/backend/core/usecases/service/SearchOffersUseCaseService.java:37]: What's wrong with this one?

Comment (242370) [reply offer/src/main/java/org/shareio/backend/core/usecases/service/SearchOffersUseCaseService.java:37]: When the offer title is "Ładny szop" we don't want it to be found by typing "SZ"

**Review by 242370 — APPROVED** [review]

**Pull request merged**